website/client/src/app.vue

```diff
@@ -609,6 +609,22 @@ export default {
 
         if (modalBefore) this.$root.$emit('bv::show::modal', modalBefore, { fromRoot: true });
       });
+
+      this.$root.$on('habitica::dismiss-modal', oldModal => {
+        if (!oldModal) return;
+        let removeIndex = this.$store.state.modalStack
+          .map(modal => modal.modalId)
+          .indexOf(oldModal);
+        if (removeIndex >= 0) {
+          this.$store.state.modalStack.splice(removeIndex, 1);
+        }
+        removeIndex = this.$store.state.modalStack
+          .map(modal => modal.prev)
+          .indexOf(oldModal);
+        if (removeIndex >= 0) {
+          delete this.$store.state.modalStack[removeIndex].prev;
+        }
+      });
     },
     validStack (modalStack) {
```

Comment:
I was wondering if we could extend this function to add the new functionality since it seems to have some overlapping

Reply:
Hmm, can you elaborate? `validStack` seems to be used to clean things up when a single modal event adds multiple copies of a modal to the stack, which isn't the problem at hand--the problem is that our `modalStack` function does exactly what it's supposed to do, but we don't actually want to return to the modals earlier in the sequence.

Reply:
got it!